test/new-e2e/pkg/utils/infra/stack_manager.go

```diff
@@ -339,45 +339,41 @@ func (sm *StackManager) getStack(ctx context.Context, name string, config runner
 	progressStreamsDestroyOption := sm.getProgressStreamsOnDestroy(logger)
 
 	var upResult auto.UpResult
+	var upError error
 	upCount := 0
 
 	for {
 		upCount++
 		upCtx, cancel := context.WithTimeout(ctx, stackUpTimeout)
-		upResult, err = stack.Up(upCtx, progressStreamsUpOption, optup.DebugLogging(loggingOptions))
+		upResult, upError = stack.Up(upCtx, progressStreamsUpOption, optup.DebugLogging(loggingOptions))
 		cancel()
 
-		if err == nil {
-			err := sendEventToDatadog(fmt.Sprintf("[E2E] Stack %s : success on Pulumi stack up", name), "", []string{"operation:up", "result:ok", fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", upCount)}, logger)
-			if err != nil {
-				fmt.Fprintf(logger, "Got error when sending event to Datadog: %v", err)
-			}
+		if upError == nil {
+			sendEventToDatadog(fmt.Sprintf("[E2E] Stack %s : success on Pulumi stack up", name), "", []string{"operation:up", "result:ok", fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", upCount)}, logger)
 			break
 		}
 
 		retryStrategy := sm.getRetryStrategyFrom(err, upCount)
-		err := sendEventToDatadog(fmt.Sprintf("[E2E] Stack %s : error on Pulumi stack up", name), err.Error(), []string{"operation:up", "result:fail", fmt.Sprintf("retry:%s", retryStrategy), fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", upCount)}, logger)
-		if err != nil {
-			fmt.Fprintf(logger, "Got error when sending event to Datadog: %v", err)
-		}
+		sendEventToDatadog(fmt.Sprintf("[E2E] Stack %s : error on Pulumi stack up", name), upError.Error(), []string{"operation:up", "result:fail", fmt.Sprintf("retry:%s", retryStrategy), fmt.Sprintf("stack:%s", stack.Name()), fmt.Sprintf("retries:%d", upCount)}, logger)
+
 		switch retryStrategy {
 		case reUp:
 			fmt.Fprint(logger, "Got error during stack up, retrying")
```

Comment:
Do we want to log the error, even if we are retrying?

Reply:
Good point, adding it in a follow up